error C2491: “OgreBites::ApplicationContextQt::staticMetaObject”

error C2491: “OgreBites::ApplicationContextQt::staticMetaObject”: dllimport static member is not allowed in moc_OgreApplicationContextQt.cpp

Windows 10 + QT 5.14.2 + VC2017 x64

--- a/CMake/Dependencies.cmake
+++ b/CMake/Dependencies.cmake
@@ -297,7 +297,8 @@ if(NOT ANDROID AND NOT EMSCRIPTEN)
     )
   endif()
 
-  find_package(Qt5 COMPONENTS Core Gui QUIET)
+  find_package(Qt5 COMPONENTS Core Gui)
+  macro_log_feature(PYTHONLIBS_FOUND "Qt" "optional integration with the Qt Library for window creation and input" "http://www.qt.io/" FALSE "" "")
 endif()
 
 #######################################################################

--- a/Components/Bites/include/OgreApplicationContextQt.h
+++ b/Components/Bites/include/OgreApplicationContextQt.h
@@ -9,6 +9,12 @@
 #include <QObject>
 #include <QWindow>
 
+#if defined(OgreBitesQt_EXPORTS)
+#  define _OgreBitesQtExport Q_DECL_EXPORT
+#else
+#  define _OgreBitesQtExport Q_DECL_IMPORT
+#endif
+
 namespace OgreBites
 {
 
@@ -21,7 +27,7 @@ Specialization for connecting with Qt
 
 Assumes that Ogre Main loop is used for compatibility with other implementations.
  */
-class _OgreBitesExport ApplicationContextQt : public QObject, public ApplicationContextBase
+class _OgreBitesQtExport ApplicationContextQt : public QObject, public ApplicationContextBase
 {
     Q_OBJECT
 public:

--- a/Components/Bites/src/OgreApplicationContextQt.cpp
+++ b/Components/Bites/src/OgreApplicationContextQt.cpp
@@ -7,6 +7,8 @@
 
 #include "OgreApplicationContextQt.h"
 
+#include <cctype>
+
 #include "OgreRoot.h"
 #include "OgreRenderWindow.h"
 

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -8,11 +8,12 @@ install:
  - cmd: cd %APPVEYOR_BUILD_FOLDER%
  - cmd: git submodule update --init --recursive
 build_script:
+ - set PATH=C:\Qt\5.12\msvc2017_64\bin;%PATH%
  - cmake -P ci-build.cmake
- - cmake --build build --config Release --target INSTALL
+ - cmake --build build --config RelWithDebInfo --target INSTALL
 test_script:
  - set PATH=%PATH%;build\gtest\lib
- - IF "%APPVEYOR_BUILD_WORKER_IMAGE%" == "Visual Studio 2017" build\bin\release\Test_Ogre.exe --gtest_filter=-MeshWithoutIndexDataTests*
+ - IF "%APPVEYOR_BUILD_WORKER_IMAGE%" == "Visual Studio 2017" build\bin\RelWithDebInfo\Test_Ogre.exe --gtest_filter=-MeshWithoutIndexDataTests*
 cache:
 - ogredeps -> CMake/Dependencies.cmake
 branches:

--- a/ci-build.cmake
+++ b/ci-build.cmake
@@ -38,11 +38,6 @@ endif()
 
 if(DEFINED ENV{APPVEYOR})
     set(CMAKE_BUILD_TYPE Release)
-    if("$ENV{APPVEYOR_BUILD_WORKER_IMAGE}" STREQUAL "Visual Studio 2017")
-        set(GENERATOR -G "Visual Studio 15")
-    else()
-        set(GENERATOR -G "Visual Studio 12")
-    endif()
     set(RENDERSYSTEMS
         -DOGRE_BUILD_RENDERSYSTEM_D3D9=TRUE
         -DOGRE_BUILD_RENDERSYSTEM_GL=TRUE
@@ -55,6 +50,15 @@ if(DEFINED ENV{APPVEYOR})
         "-DPYTHON_EXECUTABLE=C:\\Python37-x64\\python.exe"
         "-DPYTHON_LIBRARY=C:\\Python37-x64\\libs\\python37.lib"
         -DOGRE_DEPENDENCIES_DIR=${CMAKE_CURRENT_SOURCE_DIR}/ogredeps)
+
+    if("$ENV{APPVEYOR_BUILD_WORKER_IMAGE}" STREQUAL "Visual Studio 2017")
+        set(GENERATOR -G "Visual Studio 15")
+        set(OTHER ${OTHER}
+            -DCMAKE_PREFIX_PATH="C:\\Qt\\5.12\\msvc2017_64"
+            -DQt5_DIR="C:\\Qt\\5.12\\msvc2017_64\\lib\\cmake\\Qt5")
+    else()
+        set(GENERATOR -G "Visual Studio 12")
+    endif()
 
     set(BUILD_DEPS TRUE)
     set(SWIG_EXECUTABLE "C:\\ProgramData\\chocolatey\\bin\\swig.exe")

